app/client/keybase/vault.go

```diff
@@ -0,0 +1,317 @@
+		return nil, nil, nil
+	}
+
+	var addresses []string
+	var keyPairs []crypto.KeyPair
+
+	for _, key := range data.Data["keys"].([]any) {
+		addresses = append(addresses, key.(string))
+		keyPair, err := vk.Get(key.(string))
+		if err != nil {
+			return nil, nil, err
+		}
+		keyPairs = append(keyPairs, keyPair)
+	}
+
+	return addresses, keyPairs, nil
+}
+
+// Exists checks if a key exists in vault
+func (vk *vaultKeybase) Exists(address string) (bool, error) {
+	_, err := vk.Get(address)
+	if err != nil {
+		if errors.Is(err, errors.New("key not found")) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+
+// ExportPrivString exports a private key as a hex string
+func (vk *vaultKeybase) ExportPrivString(address, passphrase string) (string, error) {
+	privKey, err := vk.Get(address)
+	if err != nil {
+		return "", err
+	}
+
+	privKeyHex, err := privKey.ExportString(passphrase)
+	if err != nil {
+		return "", err
+	}
+
+	return privKeyHex, nil
+}
+
+// ExportPrivJSON exports a private key as a JSON string
+func (vk *vaultKeybase) ExportPrivJSON(address, passphrase string) (string, error) {
+	privKey, err := vk.Get(address)
+	if err != nil {
+		return "", err
+	}
+
+	privKeyJSON, err := privKey.ExportJSON(passphrase)
+	if err != nil {
+		return "", err
+	}
+
+	return privKeyJSON, nil
+}
+
+// UpdatePassphrase updates the passphrase of a key
+func (vk *vaultKeybase) UpdatePassphrase(address, oldPassphrase, newPassphrase, hint string) error {
+	privKey, err := vk.GetPrivKey(address, oldPassphrase)
+	if err != nil {
+		return err
+	}
+	privStr := privKey.String()
+
+	keyPair, err := crypto.CreateNewKeyFromString(privStr, newPassphrase, hint)
+	if err != nil {
+		return err
+	}
+
+	return writeVaultKeyPair(vk, keyPair.GetAddressString(), keyPair, hint)
+}
+
+// Sign a message using a private key
+func (vk *vaultKeybase) Sign(address, passphrase string, msg []byte) ([]byte, error) {
+	privKey, err := vk.GetPrivKey(address, passphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	sig, err := privKey.Sign(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return sig, nil
+}
+
+// Verify a message signature using a public key
+func (vk *vaultKeybase) Verify(address string, msg, sig []byte) (bool, error) {
+	pubKey, err := vk.GetPubKey(address)
+	if err != nil {
+		return false, err
+	}
+
+	return pubKey.Verify(msg, sig), nil
+}
+
+// Delete a keypair from vault
+func (vk *vaultKeybase) Delete(address, passphrase string) error {
+
+	_, err := vk.GetPrivKey(address, passphrase)
+	if err != nil {
+		return err
+	}
+
+	versionsMeta, err := vk.client.KVv2(vk.mount).GetVersionsAsList(context.TODO(), address)
+	if err != nil {
+		return err
+	}
+
+	versions := make([]int, 0, len(versionsMeta))
+	for _, version := range versionsMeta {
+		versions = append(versions, version.Version)
+	}
+
+	return vk.client.KVv2(vk.mount).Destroy(context.TODO(), address, versions)
+}
+
+func (vk *vaultKeybase) MarshalJSON() ([]byte, error) {
```

Comment:
Just curious what these 2 methods are for? Do we need to marshal the entire keybase?

Reply:
nope, just left over methods. prob messed up a rebase. should be dropped.